sensing/pointcloud_preprocessor/src/filter.cpp

```diff
@@ -162,38 +162,39 @@ void pointcloud_preprocessor::Filter::unsubscribe()
 void pointcloud_preprocessor::Filter::computePublish(
   const PointCloud2ConstPtr & input, const IndicesPtr & indices)
 {
-  PointCloud2 output;
+  auto cloud_tf = std::make_unique<PointCloud2>();
+
   // Call the virtual method in the child
-  filter(input, indices, output);
+  filter(input, indices, *cloud_tf);
```

Comment:
![image](https://user-images.githubusercontent.com/8327598/171076789-e3911b28-d04b-4c29-bf20-2cc55e9611d4.png)
`output` is already passed by reference, so it seems to be not copied.

Reply:
Sorry, if my explanation was not clear. Yes output is passed by reference so there is no problem with the `filter` function. The duplication occurs with the next line: `auto cloud_tf = std::make_unique<PointCloud2>(output);`
What my PR does is to remove this data copy by using 'cloud_tf' from the beginning

Reply:
Sorry, I missed something simple.
It is reasonable. :+1: